src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/SmoQueryModel.tt

```diff
@@ -309,6 +309,14 @@ namespace Microsoft.SqlTools.ServiceLayer.ObjectExplorer.SmoModel
 	   {
 			flags.Add("ValidForFlag.NotSqlDw");
        }
+	   if (validForStr.Contains("SqlOnDemand"))
+	   {
+			flags.Add("ValidForFlag.SqlOnDemand");
+	   }
+	   if (validForStr == "NotSqlDemand")
```

Comment:
>== [](start = 20, length = 2)

You should be doing a .Contains here

Reply:
I was using ValidForFlag.All as a roll model for this. Since it includes all other flags except Sql On Demand it seems ok to check for equality instead of .contains